### PR TITLE
Feat/users

### PR DESCRIPTION
**Pull request by 76eren**

### CODE REVIEW

File: src/app/shared/service/auth.service.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export class AuthService {
     return this.apiService.get<any>(endpoint)
       .pipe(
         map(response => {
+          console.log(response.payload)
```
Comment (isabelhb) [inline src/app/shared/service/auth.service.ts]: console log :(

File: src/app/shared/service/auth.service.ts [hunk 2]
```diff
@@ -30,9 +31,24 @@ export class AuthService {
...
-  public isAdmin(): boolean {
-    // Implement after the API is ready to handle this
-    return true;
+  public isAdmin(): Observable<boolean> {
+    let endpoint = '/auth/isAdmin';
+
+    return this.apiService.get<any>(endpoint)
+      .pipe(
+        map(response => {
+          if (response.payload && response.payload.admin) { // TODO: validate response payload
```
Comment (isabelhb) [inline src/app/shared/service/auth.service.ts]: todp & console.log

File: src/app/users/users.component.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  public searchInput: String = '';
+
+
+  constructor(private userService: UserService) {}
+
+  ngOnInit(): void {
+    this.userService.getAllUsers().subscribe(users => {
+      this.users = users;
+      this.usersToShow = users;
+      console.log(this.users)
```
Comment (isabelhb) [inline src/app/users/users.component.ts]: console.log

File: src/app/users/users.component.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    this.usersToShow = this.users.filter(user => {
+      let fullName = user.firstName.toLowerCase() + " " + user.lastName.toLowerCase();
+      let phone = user.phoneNumber.toLowerCase();
+      let search = this.searchInput.toLowerCase();
+      return user.email.toLowerCase().includes(search) || fullName.includes(search) || phone.includes(search)
+    });
+
+
+  }
+
```
Comment (isabelhb) [inline src/app/users/users.component.ts:56]: 4 enters yucky (sorry)

**Review by isabelhb — CHANGES_REQUESTED** [review]

File: src/app/shared/service/auth.service.ts (hunk 1 above)
Comment (76eren) [reply src/app/shared/service/auth.service.ts]: donny's_theme_cover

**Review by isabelhb — APPROVED** [review]

**Review by Yaman-Kwefati — APPROVED** [review]

**Pull request merged**